data-products/src/new_tab_recommendations/aggregate_engagement_flow.py

```diff
@@ -29,27 +46,71 @@
             COUNT(*) AS TRAILING_1_DAY_IMPRESSIONS,
-            SUM(CASE WHEN click IS NOT NULL THEN 1 ELSE 0 END) AS TRAILING_1_DAY_OPENS,
-            -- For now, we only need 1 day trailing data, so leave the other ones at 0.  
-            0 AS TRAILING_7_DAY_IMPRESSIONS,
-            0 AS TRAILING_7_DAY_OPENS,
-            0 AS TRAILING_14_DAY_IMPRESSIONS,
-            0 AS TRAILING_14_DAY_OPENS,
-            0 AS TRAILING_21_DAY_IMPRESSIONS,
-            0 AS TRAILING_21_DAY_OPENS,
-            0 AS TRAILING_28_DAY_IMPRESSIONS,
-            0 AS TRAILING_28_DAY_OPENS
+            SUM(CASE WHEN click IS NOT NULL THEN 1 ELSE 0 END) AS TRAILING_1_DAY_OPENS
         FROM impressions_data
         CROSS JOIN UNNEST(impressions_data.tiles) AS flattened_tiles
         GROUP BY TILE_ID
         ORDER BY TRAILING_1_DAY_IMPRESSIONS DESC
         LIMIT 16384 -- Limit to Snowflake's max list size. There are a lot of old items that still get some impressions.
-    """
+    """  # noqa: E501
+
+EXPORT_GLEAN_TELEMETRY_SQL = f"""
+DECLARE max_ts timestamp;
+
+-- create dataset
+create schema if not exists `moz-fx-mozsocial-dw-{NEW_TAB_REC_GCP_PROJECT_ENV}.{NEW_TAB_REC_DATASET}`;
+
+-- table if not exists
+create table if not exists `moz-fx-mozsocial-dw-{NEW_TAB_REC_GCP_PROJECT_ENV}.{NEW_TAB_REC_DATASET}.pocket_user_events` as (
+      SELECT
+        document_id,
+        submission_timestamp,
+        event.name AS event_name,
+        extra.value AS recommendation_id,
+        current_timestamp() AS _loaded_at
+      FROM `moz-fx-data-shared-prod.firefox_desktop.newtab_live` AS e
+      CROSS JOIN UNNEST(e.events) AS event
+      CROSS JOIN UNNEST(event.extra) AS extra ON extra.key = 'recommendation_id'
+      WHERE
+        submission_timestamp > TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL 1 DAY)
+        AND 1 = 2
+);
+-- get max submission timestamp from table
+SET max_ts = (select coalesce(max(submission_timestamp), TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL 1 DAY)) as max_ts
+from `moz-fx-mozsocial-dw-{NEW_TAB_REC_GCP_PROJECT_ENV}.{NEW_TAB_REC_DATASET}.pocket_user_events`);
+
+-- insert new records
+insert into `moz-fx-mozsocial-dw-{NEW_TAB_REC_GCP_PROJECT_ENV}.{NEW_TAB_REC_DATASET}.pocket_user_events`
+      SELECT
+        document_id,
+        submission_timestamp,
+        event.name AS event_name,
+        extra.value AS recommendation_id,
+        current_timestamp() AS _loaded_at
+      FROM `moz-fx-data-shared-prod.firefox_desktop.newtab_live` AS e
+      CROSS JOIN UNNEST(e.events) AS event
+      CROSS JOIN UNNEST(event.extra) AS extra ON extra.key = 'recommendation_id'
+      WHERE
+        submission_timestamp > max_ts
+        AND app_build >= '20231116134553' -- Fx 120 was the first build to emit recommendation_id
+        AND event.category = 'pocket'
+        AND event.name in ('click', 'impression');
```

Comment:
Just curious: why not do this incremental update in Dbt? Is Dbt not set up for BigQuery yet?

Reply:
we could but then you would to create a view to union the static and read time data.  This just felt cleaner for now....we could always add it later if there a need for others to use the source data.